helixbass/tree-sitter-grep
# Specify files

In this PR:
- (optionally) specify which files to search on the command line
- update omission of `./` in formatted match output paths to not be based on "output mode"
- update searcher/printer "architecture" to still parallelize searching/printing but hopefully not be subject to "interleaving" of printed matches

To test:
If you don't explicitly specify any files on the command line it should still search "all" files
If you specify one or more file paths as positional command-line arguments it should only search/show matches for those files
Regardless of output "mode" (ie presence or absence of `--vimgrep`) it should only include `./` in printed match output paths if you explicitly specified a corresponding file path that also included a `./`
You shouldn't see interleaved results

Performance-wise (based again on simple testing via `time`) it looks like this (presumably the searcher/printer architecture change) slowed it down just a bit, maybe from ~280-300ms -> ~290-320ms (for the same test case I was using before)

Closes #16 

Based on `test-sorted-output`
